### ISSUE TITLE
object-curly-spacing not respecting objectsInObjects when destructuring

### ISSUE THREAD

**Issue by adamrenklint** [eslint/eslint#6469]
**What version of ESLint are you using?**
2.13.0

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**
```
{
  "parser": "eslint",
  "extends": "eslint:recommended",
  "rules": {
    "object-curly-spacing": [2, "always", {
      "objectsInObjects": false,
      "arraysInObjects": true
    }]
  }
}
```

**What did you do? Please include the actual source code causing the issue.**
Wrote code like `const { y: { z }} = x`

**What did you expect to happen?**
No error to be reported

**What actually happened? Please include the actual, raw output from ESLint.**
Error was reported:
```
1:17  error  A space is required before '}'  object-curly-spacing
```

---

I forked the project to investigate, and will also open a pull request in a few minutes with a proposed fix.

**Comment by aubergine10** [comment]
Should the rule docs be updated to include `var { y: { z }} = x` in examples?

**Comment by kaicataldo** [comment]
That definitely seems counter-intuitive - I wonder if this should be lumped into objectsInObjects or not. Note that this also occurs for destructuring arrays in objects as well:
```js
const { y: [ z ] } = x;
```

**Comment by nzakas** [comment]
Confirmed, looks like a bug. We treat `ObjectPattern` and `ObjectExpression` the same in this rule so they should behave the same way throughout.
